ext/storage_sources/azure_store/test/test_azure_connection.cpp

```diff
@@ -30,5 +30,283 @@
+        CHECK(conn.read_object(file_name, -1, 1000, buffer) == -1);
+        memset(buffer, 0, 1024);
+
+        // Test overflow with negative offset.
+        CHECK(conn.read_object(file_name, -1, 12, buffer) == -1);
+        free(buffer);
+
+        blob_client.Delete();
+
+        // Tests that reading a non existant object returns a ENOENT.
+        CHECK(conn.read_object(non_exi_object_key, 0, 1, buffer) == ENOENT);
+    }
+
+    SECTION("(Test complex case).", "[azure-connection]")
+    {
+        azure_connection conn = azure_connection("myblobcontainer1", "complex_case_");
+        azure_connection conn_bad = azure_connection("myblobcontainer1", "bad_prefix_");
+
+        // Read an object in the container from middle to end
+        // Prefix for objects in this test.
+        const std::string prefix = "complex_case_";
+        const bool list_single = true;
+        void *buffer = calloc(1024, sizeof(char));
+        std::vector<std::string> objects;
+
+        // Total objects to insert in the test.
+        const int32_t num_objects = 11;
+
+        // Check that the object does not exist in the container.
+        CHECK(conn.object_exists(file_name_2, exists) == 0);
+        CHECK(exists == false);
+
+        // Put an object into the container.
+        CHECK(conn.put_object(file_name_2, path_2) == 0);
+
+        // Check that the object exists in the container.
+        CHECK(conn.object_exists(file_name_2, exists) == 0);
+        CHECK(exists == true);
+
+        // Check that the offset and length works by finding a substring in the payload.
+        CHECK(conn.read_object(file_name_2, payload_2.find(payload_2_substr),
+                payload_2_substr.length(), buffer) == 0);
+        CHECK(static_cast<char *>(buffer) == payload_2_substr);
+        memset(buffer, 0, 1024);
+
+        // Put a second object into the container.
+        CHECK(conn.put_object(file_name, path) == 0);
+
+        // There are 2 objects in the container but list single. Object size should be 1.
+        CHECK(conn.list_objects(prefix, objects, list_single) == 0);
+        CHECK(objects.size() == 1);
+
+        // Check that read works from the middle to the end.
+        CHECK(conn.read_object(file_name, payload.find(payload_substr),
+                payload.length() - payload.find(payload_substr), buffer) == 0);
+        CHECK(static_cast<char *>(buffer) ==
+          payload.substr(
+            payload.find(payload_substr), payload.length() - payload.find(payload_substr)));
+        memset(buffer, 0, 1024);
+
+        // Clean up the 2 objects.
+        CHECK(conn.delete_object(file_name) == 0);
+        CHECK(conn.delete_object(file_name_2) == 0);
+
+        // No matching objects. Object size should be 0.
+        objects.clear();
+        CHECK(conn.list_objects(prefix, objects, !list_single) == 0);
+        CHECK(objects.size() == 0);
+
+        // Put objects to prepare for test.
+        for (int i = 0; i < num_objects; i++) {
```

Comment:
np: no need for {} brackets after for loop. Implement for other instances too.

Reply:
fixed https://github.com/wiredtiger/wiredtiger/pull/8747/commits/dea0e0d6dffa7c45b696b83244162ef9bbf4291f